File: backend/src/main/java/com/warmingup/backend/domain/Appointment.java
package com.warmingup.backend.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "routine_id")
    private Routine routine;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private LocalDateTime arrivalTime;

    private int travelMinutes;

    private int bufferMinutes;

    private LocalDateTime preparationStartTime;

    private LocalDateTime departureTime;

    @Builder.Default
    @OneToMany(mappedBy = "appointment", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("itemOrder ASC")
    private List<AppointmentItem> items = new ArrayList<>();
}

File: android/app/src/main/java/com/example/warming_up/data/appointment/Appointment.kt
package com.example.warming_up.data.appointment

data class Appointment(
    val id: Long,
    val name: String,
    val preparationStartTime: String,
    val departureTime: String,
    val arrivalTime: String,
    val steps: List<AppointmentStep>,
    val checklist: List<AppointmentChecklistItem>,
)

data class AppointmentStep(
    val id: Long,
    val name: String,
    val durationMinutes: Int,
    val itemOrder: Int,
    val startTime: String,
    val endTime: String,
    val completed: Boolean,
)

data class AppointmentChecklistItem(
    val id: Long,
    val name: String,
    val itemOrder: Int,
    val completed: Boolean,
)
